### PR TITLE
Create a way to enable the New Registration System for competitions

### PR DESCRIPTION
**Pull request by FinnIckler**
Not 100% sure if this is the way to go, but I added
- A new DB Column to Competition (migration pending) that determines if the new Registration System is enabled
- A new /competitions/v2 namespace that renders the new Frontend
- A new Environment variable for the URL of the CDN where the files for the frontend are stored
- An option in the Competition Navigation to enable it, this might just make sense now when testing

If we think this a good approach I will add the migration

### CODE REVIEW

File: WcaOnRails/app/models/competition.rb [hunk 1]
```diff
@@ -660,6 +660,14 @@ def trainee_delegate_ids
     @trainee_delegate_ids || trainee_delegates.map(&:id).join(",")
   end
 
+  def enable_v2_registrations
```
Comment (gregorbg) [inline WcaOnRails/app/models/competition.rb:663]: By Ruby convention, methods that modify an existing object in-place should end with an exclamation mark

Comment (gregorbg) [inline WcaOnRails/app/models/competition.rb:663]:
```suggestion
  def enable_v2_registrations!
```

Comment (gregorbg) [reply WcaOnRails/app/models/competition.rb:663]: Usages of this method in the controller need to be updated accordingly

File: WcaOnRails/app/controllers/competitions_v2_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class CompetitionsV2Controller < ApplicationController
+  def show
+    competition_id = params[:id]
+    redirect_to competition_list_path unless competition_id.present?
```
Comment (gregorbg) [inline WcaOnRails/app/controllers/competitions_v2_controller.rb:6]:
```suggestion
    competition_id = params.require(:id)
```

Comment (gregorbg) [reply WcaOnRails/app/controllers/competitions_v2_controller.rb:6]: This will do appropriate error handling for you if the param is not present.

**Review by gregorbg — CHANGES_REQUESTED** [review]
Cosmetics

File: WcaOnRails/app/models/competition.rb (hunk 1 above)
Comment (FinnIckler) [reply WcaOnRails/app/models/competition.rb:663]: I thought ! means it returns an error

Comment (FinnIckler) [reply WcaOnRails/app/models/competition.rb]: like find! vs. find

**Comment by FinnIckler** [comment]
so I need to add something to the dev export?

**Review by gregorbg — CHANGES_REQUESTED** [review]
The diff contains the schema.rb definition for jwt_denylist. Please rebase on master to get rid of this unrelated change

**Review by gregorbg — APPROVED** [review]

**Pull request merged**